index.bs

```diff
@@ -364,11 +411,74 @@ When asked to <dfn export>report the layout shift</dfn> for an active
     1. Set |newEntry|'s <dfn attribute for=LayoutShift>hadRecentInput</dfn>
         attribute to <code>true</code> if {{LayoutShift/lastInputTime}} is less
         than 500 milliseconds in the past, and <code>false</code> otherwise.
+    1. Set |newEntry|'s <dfn attribute for=LayoutShift>sources</dfn> attribute
+        to the result of invoking the algorithm to <a>report the layout shift
+        sources</a> for |D|.
     1. <a href="https://w3c.github.io/performance-timeline/#dfn-queue-a-performanceentry">Queue the PerformanceEntry</a>
         |newEntry| object.
 
 </div>
 
+Report the layout shift sources {#sec-report-layout-shift-sources}
+------------------------------------------------------------------
+
+<div algorithm="report the layout shift sources">
+When asked to <dfn>report the layout shift sources</dfn> for an active
+{{Document}} |D|, run the following steps:
+
+1. Let |C| be an empty sequence of {{LayoutShiftAttribution}} objects.
+
+1. For each member |N| of the <a>unstable node set</a> of |D|, run these steps:
+
+    1. Let |NA| be the result of running the algorithm to
+        <a>create the attribution</a> for |N| and |D|.
+    1. If there exists any member |existing| of |C| such that the
```

Comment:
It seems that this makes more sense as the first step, as `NA` is not used if we bail out here.

Reply:
Done.